### DISCUSSION TITLE
Listbox is not closed when I click on input near it which is not part of ListBox

### DISCUSSION THREAD

**Discussion by ElenaShevchenko** [tailwindlabs/headlessui#3019] [Help]
Greetings,

I am encountering an issue with my custom Currency component. The component comprises a Listbox for selecting currency and an input field for entering the currency value.
As depicted in the provided image
 ![image](https://github.com/tailwindlabs/headlessui/assets/11269499/f3124dd5-1806-40c0-bbcd-45664719737b)
the expected behavior dictates that when the dropdown is opened, clicking on the input field should close the dropdown and shift focus to the input field since it resides outside the dropdown.

Presently, when I focus on the input field, I dynamically change its type based on business requirements using the following logic:
` type={!readOnly && isFocused ? 'number' : 'text'}`

However, upon making this type change, the dropdown fails to close as anticipated.

I suspect that the issue may be related to the code within the ListBox component, particularly the goToOption function which handles microtasks. Here's a snippet of the relevant code:

` let goToOption = useEvent((focus, id, trigger) => {
    d.dispose()
    d.microTask(() => {
      if (focus === Focus.Specific) {
        return dispatch({ type: ActionTypes.GoToOption, focus: Focus.Specific, id: id!, trigger })
      }

      return dispatch({ type: ActionTypes.GoToOption, focus, trigger })
    })
  })`
  
  Thank you

**Comment by lmossman** [comment]
I am also experiencing this issue - when I have a Listbox and a Combobox next to each other, and the Listbox options are open, if I click on the Combobox input, the Listbox options stay open while the Combobox options are opened, and they overlap each other.

@RobinMalfait is someone able to look into this bug?
